CRM/Contribute/Form/UpdateBilling.php

```diff
@@ -317,31 +317,31 @@ public function postProcess() {
       CRM_Activity_BAO_Activity::create($activityParams);
 
       // send notification
-      if ($this->_subscriptionDetails->contribution_page_id) {
+      if ($this->getSubscriptionDetails()->contribution_page_id) {
         CRM_Core_DAO::commonRetrieveAll('CRM_Contribute_DAO_ContributionPage', 'id',
-          $this->_subscriptionDetails->contribution_page_id, $value, array(
+          $this->getSubscriptionDetails()->contribution_page_id, $value, array(
             'title',
             'receipt_from_name',
             'receipt_from_email',
           )
         );
-        $receiptFrom = '"' . CRM_Utils_Array::value('receipt_from_name', $value[$this->_subscriptionDetails->contribution_page_id]) . '" <' . $value[$this->_subscriptionDetails->contribution_page_id]['receipt_from_email'] . '>';
+        $receiptFrom = '"' . CRM_Utils_Array::value('receipt_from_name', $value[$this->getSubscriptionDetails()->contribution_page_id]) . '" <' . $value[$this->getSubscriptionDetails()->contribution_page_id]['receipt_from_email'] . '>';
       }
       else {
         $domainValues = CRM_Core_BAO_Domain::getNameAndEmail();
         $receiptFrom = "$domainValues[0] <$domainValues[1]>";
       }
-      list($donorDisplayName, $donorEmail) = CRM_Contact_BAO_Contact::getContactDetails($this->_subscriptionDetails->contact_id);
+      list($donorDisplayName, $donorEmail) = CRM_Contact_BAO_Contact::getContactDetails($this->getSubscriptionDetails()->contact_id);
       $tplParams['contact'] = array('display_name' => $donorDisplayName);
 
       $tplParams = array_merge($tplParams, CRM_Contribute_Form_AbstractEditPayment::formatCreditCardDetails($processorParams));
 
       $sendTemplateParams = array(
-        'groupName' => $this->_subscriptionDetails->membership_id ? 'msg_tpl_workflow_membership' : 'msg_tpl_workflow_contribution',
-        'valueName' => $this->_subscriptionDetails->membership_id ? 'membership_autorenew_billing' : 'contribution_recurring_billing',
-        'contactId' => $this->_subscriptionDetails->contact_id,
+        'groupName' => $this->getSubscriptionDetails()->membership_id ? 'msg_tpl_workflow_membership' : 'msg_tpl_workflow_contribution',
```

Comment:
FYI - I did a bunch of work on making `groupName` obsoleter at the start of the year & I'm hoping to actively drop it in the near future (I'll put it in a dev digest so just mentioning it)

Reply:
I'm aware and looked briefly but then decided it was out of scope for this PR because it would require testing the email/messagetemplate flow as well and that seemed to get complicated